app/models/custom_label.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class CustomLabel
+  def initialize(attrs)
+    @attrs = attrs
+  end
+
+  def id
+    @attrs['id']
+  end
+
+  def display_label
+    @attrs['displayLabel']
+  end
+
+  def display_on_full_text_finder
+    @attrs['displayOnFullTextFinder']
+  end
+
+  def display_on_publication_finder
+    @attrs['displayOnPublicationFinder']
+  end
+
+  def self.all
+    rmapi_customer_root.all.labels.map { |attrs| new attrs }
+  end
+
+  def self.update(label_id, attrs)
+    id = label_id.to_i
+    return unless id.positive?
+    root = rmapi_customer_root.all
+    # prune the root to not include labels with displayLabel=""
+    # because RM API would throw an error in PUT request
+    prune_root(root, id)
```

Comment:
I'd move all of the pruning / patching / updating into the `RmapiCustomerRoot` since that is the class that ultimatley has knowledge of the RMAPI and abstracts away how it works. In other words, if the RMAPI changes its contract, then only this class should need to change.

So here we would say:
```ruby
root.update_label(label_id, attrs)
return new attrs
```
and then in `RmApiCustomerRoot`

```ruby
def update_label(label_id, attrs)
  prune(label_id)
  patch(label_id, attrs)
  update()
end
```

Reply:
Sure, thanks for your suggestion! I will move it..